### PR TITLE
Add HnrPvt [DEVINFRA-877]

### PR DESCRIPTION
**Pull request by adrian-kong**
https://content.u-blox.com/sites/default/files/products/documents/u-blox8-M8_ReceiverDescrProtSpec_UBX-13003221.pdf

Page 303

### CODE REVIEW

**Comment by adrian-kong** [comment]
Works with `cat ... | cargo run` outputs

```
HnrPvt(HnrPvt { 
i_tow: 261328600, 
year: 2019, 
month: 11, 
day: 27, 
hour: 0, 
min: 35, 
sec: 10, 
valid: 55, 
nano: 599999997, 
gps_fix: 4, 
flags: GPS_FIX_OK | WKN_SET | TOW_SET | HEAD_VEH_VALID, 
reserved1: [117, 23], 
lon: -122.4016034, 
lat: 37.7930315, 
height: -29300, 
height_msl: 408, 
g_speed: 0, 
speed: 0, 
head_mot: 82.29772000000001, 
head_veh: 82.29772000000001, 
h_acc: 2406, 
v_acc: 810, 
s_acc: 78, 
head_acc: 1.09745, 
reserved2: [0, 8, 0, 12] 
})
```

File: ublox/src/ubx_packets/packets.rs [hunk 1]
```diff
@@ -1986,6 +1986,50 @@ struct EsfMeas {
...
+#[ubx(class = 0x28, id = 0x00, fixed_payload_len = 72)]
+struct HnrPvt {
+    i_tow: u32,
+    year: u16,
+    month: u8,
+    day: u8,
+    hour: u8,
+    min: u8,
+    sec: u8,
+    valid: u8,
```
Comment (notoriaga) [inline ublox/src/ubx_packets/packets.rs:2001]: I think this should also be a bitfield with values `FullyResolved`/`ValidTime`/`ValidDate`

File: ublox/src/ubx_packets/packets.rs [hunk 2]
```diff
@@ -1986,6 +1986,50 @@ struct EsfMeas {
...
+    nano: i32,
+    gps_fix: u8,
```
Comment (notoriaga) [inline ublox/src/ubx_packets/packets.rs:2004]: This one isn't marked as a bit field in the spec but it seems like it kind of is? I'ts a u8 that has 6 valid values and 2 reserved bits. Not sure if the ubx macro has something for this

Comment (adrian-kong) [reply ublox/src/ubx_packets/packets.rs:2004]: It doesn't look like a bit field, just an enum type with values 0 to 6 so not 2^8 ? can map these constants into enums with `map_type` or something

Comment (adrian-kong) [reply ublox/src/ubx_packets/packets.rs:2004]: NVM it's already implemented enum GpsFix

Comment (notoriaga) [reply ublox/src/ubx_packets/packets.rs:2004]: Ah nice

**Review by notoriaga — APPROVED** [review]

**Pull request merged**